### ISSUE TITLE
right of way tagging

### ISSUE THREAD

**Issue by RubenKelevra** [westnordost/StreetComplete#1062]
Currently, one important information of our street infrastructure is nearly completely unmapped: The right of way.

Somewhat important for routers, since they can estimate the travel time much better when this information is present. Also, it improves the audible directions since you don't have to "turn" when a priority road is turning e.g. "turning" left and a different road is going right with the same importance.

I have actually no real "plan" how to solve this, but in the last time, we did some very productive discussions in the tickets, like in the building-type ticket.

Tags currently in use for the right of way:

- stop-signs as node in the road with forward/backward tagging.
- give_way-signs as node in the road with forward/backward tagging.
- stop-signs in the intersecting node with direction=all for a "for way stop" (USA specific)
- priority_road=designated as a tag on the ways

There is also a draft for a give_way relation:

https://wiki.openstreetmap.org/wiki/Relations/Proposed/Give_way

I would like to hear some other opinions on that topic, since the current approach: Taking pictures and map that information at home requires a huge amount of time, instead of a simple SC-tag change on the go.

I wonder if we're able to come up at least with a usable solution to preselect a possible priority_road and ask the users if those selection is correct. On the other hand - I have no clue how to tag a road, if it's not the priority road :/

**Comment by rugk** [comment]
> The right of way.

The what?

**Comment by RubenKelevra** [comment]
Well, in Germany it's called Vorfahrt, my dictionary list "right of way" and "priority in traffic" for this term.

**Comment by westnordost** [comment]
Vorfahrt beachten.

Discussions about openstreetmap tagging here in the tickets, constructive as they may be, must be held in one of the official osm channels to have any weight.

**Comment by matkoniecz** [comment]
> I would like to hear some other opinions on that topic, since the current approach

Why you started discussion here, rather than on the OSM wiki or tagging mailing list?

It is not fixable by making changes in this app, so issue tracker is wrong place to discuss it.

**Comment by RubenKelevra** [comment]
Well, a discussion in OSM Wiki or Tagging mailing list does not account for the constrained and implementation of this app. I like to hear some other opinions before I ask on the mailing list, to get a complete picture of this problem.

A thing which we might be able to implement is actually the proposed Give_way relation. It wouldn't be easy, but possible since we can check if the intersection is already part of such a relation.

Do you think, this would be the best solution?

**Comment by matkoniecz** [comment]
> Do you think, this would be the best solution?

I am subscribed to tagging mailing list and I am active there. I will not participate in an offtopic discussion here.

**Comment by RubenKelevra** [comment]
@matkoniecz I think we should first think about getting an idea if it's even possible to implement and cover all cases, before starting a discussion on the actual tagging on the mailing list, of the missing cases. Or do you think we should do StreetComplete specific details on the tagging-mailing list as well?

**Comment by RubenKelevra** [comment]
Well, I hope someones else give me an opinion if this is even worth a discussion on the mailing list. I like to repeat, that I don't want to discuss any tag change here, that would be stupid. So just ignore the stubborn behavoir of @matkoniecz here, please. Thanks.

**Comment by RubenKelevra** [comment]
I don't get you. I just want to know if the implementation is worth considering an actual tagging discussion on the mailing list. I don't want to discuss a tag change of OSM here, I just want to get some opinions if an implementation is actually possible.

**Comment by RubenKelevra** [comment]
So from the comments previously written I assume, it's possible to implement a solution which would add a tagging scheme to any case possible, on simple intersections AND that there's a way to just the simple one-node-intersections to a quest.

So I started a thread on the tagging mailing list to get more people involved in the actual tagging of those non-defined cases and if it's reasonable to ditch the priority_road=designated in favor of the relation for stop, priority_road and yield (give_way) information.

https://lists.openstreetmap.org/pipermail/tagging/2018-May/036195.html

**Comment by rugk** [comment]
To see  whether a tagging scheme would work in SC, [see the wiki](https://github.com/westnordost/StreetComplete/wiki/Adding-new-Quests-to-StreetComplete#1-decide-whether-the-idea-works-out-with-streetcomplete). There is an extensive guide on how/what quests work and which won't. I think if you remember this, you can propose a tagging scheme that can be used by SC.
Generally I think it should be possible to do it as a SC quest though.

So I think you can close this issue here. You can open a new one and fill out the issue template, if the  tagging scheme is clear.

**Comment by RubenKelevra** [comment]
Well, I have gone forward with my proposal(s) an I doubt that this is currently solvable by SC, after reading this Wiki page. There's no way this can be solved without a relation.

The reason even bothered to start this is, that it's currently so hard to parse and use that information, that it's basically ignored by everyone. Other more detailed information like turn restrictions and lane tagging (where I worked the last time on) are much more usable because the information can be more easily parsed.

**Comment by rugk** [comment]
>  so hard to parse and use that information, that it's basically ignored by everyone. 

Because it is mapped with nodes on the streets, yeah… mhh…
So sounds like the tagging scheme really needs an improvement. So what's it with your proposal? Was not that an idea to make it better, so that it can also actually used by SC?

**Comment by RubenKelevra** [comment]
> So what's it with your proposal? 

The only solution which is able to map every case would be a relation holding all pieces of information together. I'm still working on that. 

> Was not that an idea to make it better, so that it can also actually used by SC?

To remain a simple approach for simple cases I will also propose some simple tags on the intersecting node on a simple intersection. But this would be pretty hard to distinguish from non-simple cases - if SC would be able to, simple 4-way-stops and right-before-left intersections should be able to solve in SC if my proposal gets accepted. But complex cases require the creation of elements: A relation at least - which isn't supported by SC at the moment.
